File: 10-19/Problem_10/problem10.c
#include <stdio.h>

int main() {
    int numbers[2000000];
    int multiple;
    unsigned long result = 0;

    for (int n=2; n < 2000000; n++) {
        if (numbers[n] < 0) continue;

        result += n;
        multiple = n*2;
        while (multiple < 2000000) {
            numbers[multiple] = -1;
            multiple += n;
        }
    }

    printf("%lu\n", result);
    return 0;
}

File: 10-19/Problem_10/problem10.py

numbers = [i for i in range(2_000_000)]
# do not consider multiples of 0 and 1
numbers[0], numbers[1] = None, None
primes = []
for n in numbers:
    if n is None: continue

    # Is prime
    primes.append(n)
    multiple = n
    while multiple+n < 2_000_000:
        numbers[multiple+n] = None
        multiple += n

print(sum(primes))